addons/web/static/tests/core/utils/strings.test.js

```diff
@@ -137,3 +138,12 @@ test("isNumeric", () => {
 
     expect(isNumeric("1234")).toBe(true);
 });
+
+test("odoomark", () => {
+    expect(odoomark("")).toBe("");
+    expect(odoomark("**test**")).toBe("<b>test</b>");
+    expect(odoomark("--test--")).toBe("<span class='text-muted'>test</span>");
```

Comment:
```suggestion
    expect(odoomark("--test--")).toBe("<span class='text-muted'>test</span>");
    expect(odoomark("--test-- something else --another test--")).toBe("<span class='text-muted'>test</span> something else <span class='text-muted'>another test</span>");
```
which doesn't work for now, but will once you'll use the lazy quantifiers as wil suggested